src/libsystemd/sd-event/sd-event.c

```diff
@@ -1799,7 +1799,7 @@ static void event_free_inode_data(
         if (!d)
                 return;
 
-        assert(!d->event_sources);
+        assert(LIST_IS_EMPTY(d->event_sources));
```

Comment:
I don't think `LIST_IS_EMPTY` is useful. We could drop it altogether, but that's a discussion for a separate PR. If we have it, it's fine to use it here.